erpnext/stock/doctype/delivery_trip/delivery_trip.py

```diff
@@ -261,9 +261,21 @@ def get_delivery_window(customer=None):
 		return delivery_window
 
 	default_window = frappe.db.get_value("Delivery Settings", "Delivery Settings", ["delivery_start_time", "delivery_end_time"], as_dict=1)
-	default_window.default_window = True
-	return default_window
+	if default_window.delivery_start_time and default_window.delivery_end_time:
+		print("Default WIndow",default_window)
+		default_window.default_window = True
+		return default_window
+	else:
+		frappe.throw(_("Default Delivery Window not available"))
 
+@frappe.whitelist()
+def get_delivery_window_from_delivery_note(delivery_note=None,customer=None):
```

Comment:
You should use the same logic as defined above, let's keep it consistent, just add the `delivery_note` check there and remove this function (this will never fallback to the global default in Delivery Settings).

Maybe define the function to take in any transaction, as:

```python
def get_delivery_window(doctype=None, docname=None, customer=None):
	# check for the passed details
```